shopgate/pwa
# Extended registration form logic and features

# Description

This ticket adds support of the customer attribute type "callingNumber". Since the FormBuilder already supported fields with phone numbers, it just extends the "account" form helpers with a mapping for the input type. 

Additionally it fixes an issue that the "supported countries" from the shop settings are not considered within the country picker of the phone input. The counties within this list are now added as a special group at the top of the country list.

Besides that it adds support for "displayMessage" property from server side form validation. That enables to provide extended validation messages for e.g. complex password rules enforced by the ecp.

## Type of change


## How to test it

- connect to a shop that has a customer attribute of type "callingNumber" configured
- navigate to the registration form and check if the field is visible and behaves as expected
- navigate to the profile screen of your newly created user and check if the field is also visible there and behaves as expected
